src/app/services/firestore.service.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Firebase App (the core Firebase SDK) is always required and
+// must be listed before other Firebase SDKs
+import * as firebase from "firebase/app";
+
+// Add the Firebase services that you want to use
+import "firebase/auth";
+import "firebase/firestore";
+import { StyleRule } from '../services/styles.service';
+
+const SHARING_COLLECTION = 'GeoVizSharing';
+
+export interface ShareableData {
+  projectId : string | undefined;
+  jobId : string | undefined;
+  location : string | undefined;
+  styles: Array<StyleRule>;
+}
+
+/**
+ * Utility class for managing interaction with the Firestore.
+ */
+export class FirestoreService {
+  private db: firebase.firestore.Firestore = null;
+
+  constructor() {
+    // Your web app's Firebase configuration
+    var firebaseConfig = {
+      apiKey: "AIzaSyDS8k-x7L9vZ_mvvdyTzwQ1LNXsYLNnhOM",
+      authDomain: "bigquerygeoviz.firebaseapp.com",
+      databaseURL: "https://bigquerygeoviz.firebaseio.com",
+      projectId: "bigquerygeoviz",
+      storageBucket: "bigquerygeoviz.appspot.com",
+      messagingSenderId: "419125973937",
+      appId: "1:419125973937:web:eba1c63d64b58be3ec2390",
+      measurementId: "G-FNH2K1BP5G"
+    };
+    // Initialize Firebase
+    firebase.initializeApp(firebaseConfig);
+    this.db = firebase.firestore();
+  }
```

Comment:
Would this auth pattern work?

https://github.com/msukmanowsky/gapi-firebase#background

Reply:
I think it worked :) 

I updated the rules to allow only logged in users.